### DISCUSSION TITLE
toggle window fix monitor

### DISCUSSION THREAD

**Discussion by The-Lost-Light** [Aylur/ags#435] [Q&A] [answered]
I want fix the window at the same monitor when its visible
and i use this code test
```ts
const hyprland = await Service.import("hyprland");

import Logout from "./modules/logout";
import Poweroff from "./modules/poweroff";
import Reboot from "./modules/reboot";
import Sleep from "./modules/sleep";

let visible = Variable(false);
let x = Utils.merge(
  [visible.bind(), hyprland.active.monitor.bind("id")],
  (v, id) => {
    print(v);
    return id;
  },
);

export default () =>
  Widget.Window({
    name: "power_menu",
    class_name: "power_menu",
    visible: visible.bind(),
    anchor: ["right"],
    monitor: x,
    child: Widget.Box({
      vertical: true,
      children: [Logout(), Poweroff(), Reboot(), Sleep()],
    }),
  });
```
and why the print is always display false
My original thought is when its invisible use 1 variable store and its return this variable when the window is toggle

Ps: Is there method i click other application except ags window then hide the ags window

**Comment by kotontrion** [comment]
1. You don't  need x, you can just bind the monitor id directly:
    ```js
    monitor: hyprland.active.monitor.bind("id"),
    ```
2. Do you even change the value of the visible Variable anywhere? 
3. If you implement point 1, you don't even need the visible Variable.

**Reply by The-Lost-Light** [reply]
I konw i can use 1.
but i want it can be fix at the same monitor when the window is visable

**Reply by kotontrion** [reply, marked as answer]
If you want to the window to open at the currently focused monitor and stay there until it is closed, you should not specify the monitor property at all. So it will switch monitors only when the window is opened.

**Reply by The-Lost-Light** [reply]
Oh thank you, I didn't expect it to be so simple at all

For another question can it hide when click on other app?

**Reply by kotontrion** [reply]
> For another question can it hide when click on other app?

Not directly.
I don't have that feature in my config, i close the windows using the escape key.
others achieve that by having a transparent eventbox around your window, which when clicked will close the window. I think Aylurs config has that, so you might want to look there.

**Reply by The-Lost-Light** [reply]
ok I'll try it!
